File: src/test/java/com/freya02/botcommands/test/BasicSettingsProvider.java
package com.freya02.botcommands.test;

import com.freya02.botcommands.api.core.SettingsProvider;
import com.freya02.botcommands.api.core.annotations.BService;
import com.freya02.botcommands.api.core.annotations.ServiceType;
import net.dv8tion.jda.api.entities.Guild;
import net.dv8tion.jda.api.entities.User;
import net.dv8tion.jda.api.interactions.DiscordLocale;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@BService
@ServiceType(types = SettingsProvider.class)
public class BasicSettingsProvider implements SettingsProvider {
	@Override
	@NotNull
	public DiscordLocale getLocale(@Nullable Guild guild) {
		if (guild != null) {
			if (guild.getIdLong() == 722891685755093072L) {
				return DiscordLocale.ENGLISH_UK; //not default on my system
			}
		}

		return SettingsProvider.super.getLocale(guild);
	}

	@Override
	public boolean doesUserConsentNSFW(@NotNull User user) {
		return true;
	}
}

File: src/test/kotlin/io/github/freya022/botcommands/test_kt/BasicSettingsProvider.kt
package io.github.freya022.botcommands.test_kt

import io.github.freya022.botcommands.api.core.SettingsProvider
import io.github.freya022.botcommands.api.core.service.annotations.BService
import net.dv8tion.jda.api.entities.Guild
import net.dv8tion.jda.api.entities.User
import net.dv8tion.jda.api.interactions.DiscordLocale

@BService
class BasicSettingsProvider : SettingsProvider {
    override fun getLocale(guild: Guild?): DiscordLocale {
        if (guild?.idLong == 722891685755093072L) {
            return DiscordLocale.ENGLISH_UK //not default on my system
        }

        return super.getLocale(guild)
    }

    override fun doesUserConsentNSFW(user: User): Boolean {
        return true
    }
}